src/bmeg/models/vertex_models.py

```diff
@@ -18,6 +18,11 @@ def dump(self):
         data = dict(self.__dict__)
         del data["gid"]
 
+        # delete null values
+        remove = [k for k in data if data[k] == None]
+        for k in remove:
+            del data[k]
```

Comment:
I'm not sure this applies to all types, and I'd like to see these models have as little logic as possible. What's the specific case you're applying to, and can it live in a transformer?

Reply:
@buchanae : thanks for the thorough review.  we should have more of them... 💯 

If the model has:
``` my_property: str = None```
and it is never set, the resulting json will be 
``` "my_property": null ```
pretty noisy.

I'd prefer the transformers not delve into serialization issues?
How about an optional parameter that defaults to False
e.g.
```
    def emit(self, obj, preserve_nulls=False):
        label = obj.__class__.__name__

        if isinstance(obj, Vertex):
            suffix = "Vertex"
        elif isinstance(obj, Edge):
            suffix = "Edge"
        else:
            raise TypeError("emit accepts objects of the Vertex or Edge type")

        fname = "%s.%s.%s.json" % (self.prefix, label, suffix)

        if fname in self.handles:
            fh = self.handles[fname]
        else:
            fh = open(fname, "w")
            self.handles[fname] = fh

        fh.write(obj.dump(preserve_nulls))
        fh.write(os.linesep)
        return
```